src/cryptonote_protocol/quorumnet.cpp

```diff
@@ -0,0 +1,1435 @@
+    for (size_t qi = 0; qi < blink_quorums.size(); qi++)
+        btx.limit_signatures(static_cast<blink_tx::subquorum>(qi), blink_quorums[qi]->validators.size());
+
+    {
+        crypto::hash tx_hash_actual;
+        if (!cryptonote::parse_and_validate_tx_from_blob(tx_data, tx, tx_hash_actual)) {
+            MINFO("Rejecting blink tx: failed to parse transaction data");
+            if (tag)
+                m.reply("bl_nostart", bt_dict{{"!", tag}, {"e", "Failed to parse transaction data"}});
+            return;
+        }
+        MTRACE("Successfully parsed transaction data");
+
+        if (tx_hash != tx_hash_actual) {
+            MINFO("Rejecting blink tx: submitted tx hash " << tx_hash << " did not match actual tx hash " << tx_hash_actual);
+            if (tag)
+                m.reply("bl_nostart", bt_dict{{"!", tag}, {"e", "Invalid transaction hash"}});
+            return;
+        } else {
+            MTRACE("Pre-computed tx hash matches actual tx hash");
+        }
+    }
+
+    // Abort if we don't have at least one strong peer to send it to.  This can only happen if it's
+    // a brand new SN (not just restarted!) that hasn't received uptime proofs before.
+    if (!pinfo.strong_peers) {
+        MWARNING("Could not find connection info for any blink quorum peers.  Aborting blink tx");
+        if (tag)
+            m.reply("bl_nostart", bt_dict{{"!", tag}, {"e", "No quorum peers are currently reachable"}});
+        return;
+    }
+
+    // See if we've already handled this blink tx, and if not, store it.  Also check for any pending
+    // signatures for this blink tx that we received or processed before we got here with this tx.
+    std::list<pending_signature> signatures;
+    {
+        std::unique_lock<std::shared_timed_mutex> lock(snw.mutex);
+        auto &bl_info = snw.blinks[blink_height][tx_hash];
+        if (bl_info.btxptr) {
+            MDEBUG("Already seen and forwarded this blink tx, ignoring it.");
+            return;
+        }
+        bl_info.btxptr = btxptr;
+        for (auto &sig : bl_info.pending_sigs)
+            signatures.push_back(std::move(sig));
+        bl_info.pending_sigs.clear();
+        if (tag > 0) {
+            bl_info.reply_tag = tag;
+            bl_info.reply_pubkey = m.pubkey;
+        }
+    }
+    MTRACE("Accepted new blink tx for verification");
+
+    // The submission looks good.  We distribute it first, *before* we start verifying the actual tx
+    // details, for two reasons: we want other quorum members to start verifying ASAP, and we want
+    // to propagate to peers even if the things below fail on this node (because our peers might
+    // succeed).  We test the bits *above*, however, because if they fail we won't agree on the
+    // right quorum to send it to.
+    //
+    // FIXME - am I 100% sure I want to do the above?  Verifying the TX would cut off being able to
+    // induce a node to broadcast a junk TX to other quorum members.
```

Comment:
After some thought, I feel like we want this. This seems too abusable. It sort of work outs if you just do handle_incoming_tx and set do_not_relay to true.

Then your added set_relayable() to make it relayable once the signatures have arrived.

Currently calling add_tx directly doesn't check if the tx exists in the blockchain yet (but going through core handle_incoming_tx does, though that appears to be the only check missing from just going straight to add_tx in txpool)